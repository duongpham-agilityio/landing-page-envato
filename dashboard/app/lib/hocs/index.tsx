export { withBalance } from './withBalance';
export { withSendMoney } from './withSendMoney';
export { withAddMoney } from './withAddMoney';
export { withPinCode } from './withPinCode';
