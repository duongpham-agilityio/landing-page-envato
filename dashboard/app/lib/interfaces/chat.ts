export interface TChat {
  messages: string;
  isSend: boolean;
  uid: string;
  content: string;
}

export type TMessages = {
  date: { nanoSeconds: number; second: number };
  id: string;
  senderId: string;
  text: string;
};
