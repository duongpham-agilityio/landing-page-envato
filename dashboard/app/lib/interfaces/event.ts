export interface TEvent {
  _id: string;
  eventName: string;
  startTime: string;
  endTime: string;
}
