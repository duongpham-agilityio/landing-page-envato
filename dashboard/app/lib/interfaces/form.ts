import { TUserDetail } from './user';

export type TPinCodeForm = {
  pinCode: string;
  userId?: string;
};

export type TAuthForm = Omit<TUserDetail, 'id' | 'createdAt'> & {
  confirmPassword: string;
  isAcceptPrivacyPolicy: boolean;
  isRemember: boolean;
};

export type TAddMoneyForm = {
  userId: string;
  amount: string;
};
