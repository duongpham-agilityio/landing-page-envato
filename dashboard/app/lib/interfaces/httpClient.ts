// Libs
import { AxiosInstance } from 'axios';

export type TOnActivity = (
  httpService: AxiosInstance,
  actionName?: string,
  userId?: string,
) => number;

export type QueryOptions = {
  path: string;
  data: object;
  configs?: object;
  userId?: string;
  searchParam?: string;
  page?: number;
  limit?: number;
  actionName?: string;
  onActivity?: TOnActivity;
};
