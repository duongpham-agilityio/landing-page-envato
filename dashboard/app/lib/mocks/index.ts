export * from './spending';
export * from './table';
export * from './transaction';
export * from './user';
export * from './time';
export * from './statistic';
export * from './chat';
export * from './issues';
export * from './wallet';
