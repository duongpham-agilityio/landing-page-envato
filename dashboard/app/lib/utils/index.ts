export * from './transaction';
export * from './time';
export * from './user';
export * from './url';
export * from './helpers';
export * from './text';
export * from './localStorage';
export * from './image';
export * from './prefetch';
export * from './toast';
