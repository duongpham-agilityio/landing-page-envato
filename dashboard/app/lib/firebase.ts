// Firebase
import {
  Timestamp,
  arrayUnion,
  doc,
  serverTimestamp,
  updateDoc,
} from 'firebase/firestore';

// Utils
import { db } from './utils';

// Interface
import { TMessages } from './interfaces';

export const adminSendMessage = async (
  data: TMessages,
  idRoomChat: string,
  adminId: string,
) => {
  await updateDoc(doc(db, 'chats', idRoomChat), {
    messages: arrayUnion({
      id: data.id,
      text: data.text,
      senderId: data.senderId,
      date: Timestamp.now(),
    }),
  });

  await updateDoc(doc(db, 'userChats', adminId), {
    [idRoomChat + '.lastMessage']: {
      text: data.text,
    },
    [idRoomChat + '.date']: serverTimestamp(),
  });
};
