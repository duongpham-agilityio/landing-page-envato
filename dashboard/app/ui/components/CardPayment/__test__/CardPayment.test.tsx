// Libs
import { screen } from '@testing-library/react';
import { useForm } from 'react-hook-form';

// Types
import { TTransfer } from '@/lib/interfaces';

// Utils
import { renderQueryProviderTest } from '@/lib/utils';

// Mocks
import { MOCK_FILTER_DATA_USERS } from '@/lib/mocks';

// Components
import { CardPayment } from '..';

describe('CardPayment', () => {
  const { result } = renderHook(() => useForm<TTransfer>());

  const mockDirtyFields = {
    amount: true,
    memberId: true,
  };

  const mockProps = {
    control: result.current.control,
    dirtyFields: mockDirtyFields,
    userList: MOCK_FILTER_DATA_USERS,
    isSendMoneySubmitting: true,
    onSubmitSendMoneyHandler: result.current.handleSubmit,
    onSubmitSendMoney: jest.fn(),
  };

  it('should match with snapshot', () => {
    const { container } = renderQueryProviderTest(
      <CardPayment {...mockProps} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('The Send button should be enabled when the form is dirty', () => {
    renderQueryProviderTest(
      <CardPayment {...mockProps} isSendMoneySubmitting={false} />,
    );

    const sendButton = screen.getByText('Send Money');
    expect(sendButton).not.toBeDisabled();
  });
});
