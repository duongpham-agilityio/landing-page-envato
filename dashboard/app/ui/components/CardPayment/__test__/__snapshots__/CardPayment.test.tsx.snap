// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPayment should match with snapshot 1`] = `
<div>
  <div
    class="css-1p0sw78"
  >
    <h3
      class="chakra-heading css-1jhdndu"
    >
      my wallet
    </h3>
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-140b0vm"
      >
        <div
          class="css-1loy0ln"
        >
          <p
            class="chakra-text css-mskrhv"
          >
            Balance
          </p>
          <button
            aria-label="eye"
            class="chakra-button css-he8p55"
            data-testid="btn-eye"
            type="button"
          >
            <svg
              height="16"
              viewBox="0 0 576 512"
              width="18"
            >
              <g
                opacity="0.6"
              >
                <path
                  d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
                  fill="white"
                  opacity="1"
                />
              </g>
            </svg>
          </button>
        </div>
        <p
          class="chakra-text css-llqfbq"
        >
          ******
        </p>
      </div>
    </div>
    <form
      class="css-1ayfwcb"
    >
      <p
        class="chakra-text css-y05pe7"
      >
        quick transfer
      </p>
      <div
        class="css-79elbk"
      >
        <div
          class="chakra-input__group css-1lq0mp5"
          data-group="true"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-18ql1xp"
                id="field-:r4:"
                name="memberId"
                placeholder="Choose an account to transfer"
                type="text"
                value=""
              />
              <button
                aria-label="The eye icon"
                class="chakra-input__right-element css-1stj06r"
                data-testid="right-icon-input"
                type="button"
              >
                <button
                  aria-label="Icon dropdown"
                  class="chakra-button css-1rb8ldf"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="21"
                    style="transform: translateY(4px);"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                      stroke="#A0AEC0"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </button>
            </div>
          </div>
          <div
            class="chakra-input__right-element css-f46tp1"
          >
            <button
              aria-label="Icon dropdown"
              class="chakra-button css-1rb8ldf"
              type="button"
            >
              <svg
                fill="none"
                height="21"
                style="transform: translateY(4px);"
                viewBox="0 0 21 21"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                  stroke="#A0AEC0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1hlzswh"
      >
        <p
          class="chakra-text css-1ki0mgu"
        >
           Enter amount 
        </p>
        <div
          class="css-17piex8"
        >
          <p
            class="chakra-text css-4xj0bk"
          >
            $
          </p>
          <div
            class="chakra-form-control css-1eor3tg"
            role="group"
          >
            <input
              autocomplete="off"
              class="chakra-input css-cz5fpu"
              id="field-:r5:"
              name="amount"
              placeholder="0.00"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        aria-label="btn-send-money"
        class="chakra-button css-1mu22hy"
        disabled=""
        type="submit"
      >
        Send Money
      </button>
    </form>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;

exports[`CardPayment test cases CardPayment component renders correctly 1`] = `
<div>
  <div
    class="css-9ft5fw"
  >
    <h3
      class="chakra-heading css-1vczax5"
    >
      my wallet
    </h3>
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-159ui0o"
      >
        <div
          class="css-1foj61n"
        >
          <p
            class="chakra-text css-rxozpz"
          >
            Balance
          </p>
          <button
            aria-label="eye"
            class="chakra-button css-c713z0"
            data-testid="btn-eye"
            type="button"
          >
            <svg
              height="16"
              viewBox="0 0 640 512"
              width="20"
            >
              <g
                opacity="0.6"
              >
                <path
                  d="M38.8 5.1C28.4-3.1 13.3-1.2 5.1 9.2S-1.2 34.7 9.2 42.9l592 464c10.4 8.2 25.5 6.3 33.7-4.1s6.3-25.5-4.1-33.7L525.6 386.7c39.6-40.6 66.4-86.1 79.9-118.4c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C465.5 68.8 400.8 32 320 32c-68.2 0-125 26.3-169.3 60.8L38.8 5.1zm151 118.3C226 97.7 269.5 80 320 80c65.2 0 118.8 29.6 159.9 67.7C518.4 183.5 545 226 558.6 256c-12.6 28-36.6 66.8-70.9 100.9l-53.8-42.2c9.1-17.6 14.2-37.5 14.2-58.7c0-70.7-57.3-128-128-128c-32.2 0-61.7 11.9-84.2 31.5l-46.1-36.1zM394.9 284.2l-81.5-63.9c4.2-8.5 6.6-18.2 6.6-28.3c0-5.5-.7-10.9-2-16c.7 0 1.3 0 2 0c44.2 0 80 35.8 80 80c0 9.9-1.8 19.4-5.1 28.2zm51.3 163.3l-41.9-33C378.8 425.4 350.7 432 320 432c-65.2 0-118.8-29.6-159.9-67.7C121.6 328.5 95 286 81.4 256c8.3-18.4 21.5-41.5 39.4-64.8L83.1 161.5C60.3 191.2 44 220.8 34.5 243.7c-3.3 7.9-3.3 16.7 0 24.6c14.9 35.7 46.2 87.7 93 131.1C174.5 443.2 239.2 480 320 480c47.8 0 89.9-12.9 126.2-32.5zm-88-69.3L302 334c-23.5-5.4-43.1-21.2-53.7-42.3l-56.1-44.2c-.2 2.8-.3 5.6-.3 8.5c0 70.7 57.3 128 128 128c13.3 0 26.1-2 38.2-5.8z"
                  fill="white"
                  opacity="1"
                />
              </g>
            </svg>
          </button>
        </div>
        <p
          class="chakra-text css-1was8s"
        >
          ******
        </p>
      </div>
    </div>
    <form
      class="css-1fhgjcy"
    >
      <p
        class="chakra-text css-1vczax5"
      >
        quick transfer
      </p>
      <div
        class="css-79elbk"
      >
        <div
          class="chakra-input__group css-pbb1q5"
          data-group="true"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-10kn41x"
                id="field-:r7:"
                name="memberId"
                placeholder="Choose an account to transfer"
                type="text"
                value=""
              />
              <button
                aria-label="The eye icon"
                class="chakra-input__right-element css-1uzzgp1"
                data-testid="right-icon-input"
                type="button"
              >
                <button
                  aria-label="Icon dropdown"
                  class="chakra-button css-1gpu7bx"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="21"
                    style="transform: translateY(4px);"
                    viewBox="0 0 21 21"
                    width="21"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                      stroke="#A0AEC0"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </button>
            </div>
          </div>
          <div
            class="chakra-input__right-element css-dp9r5f"
          >
            <button
              aria-label="Icon dropdown"
              class="chakra-button css-1gpu7bx"
              type="button"
            >
              <svg
                fill="none"
                height="21"
                style="transform: translateY(4px);"
                viewBox="0 0 21 21"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.72168 5.03271L12.7217 10.0327L17.7217 5.03271"
                  stroke="#A0AEC0"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-qusax6"
      >
        <p
          class="chakra-text css-jz5bk3"
        >
           Enter amount 
        </p>
        <div
          class="css-fw0l5r"
        >
          <p
            class="chakra-text css-vax188"
          >
            $
          </p>
          <div
            class="chakra-form-control css-1mgooai"
            role="group"
          >
            <input
              autocomplete="off"
              class="chakra-input css-xgmxbo"
              id="field-:r8:"
              name="amount"
              placeholder="0.00"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        aria-label="btn-send-money"
        class="chakra-button css-1dj84lh"
        disabled=""
        type="submit"
      >
        Send Money
      </button>
    </form>
  </div>
</div>
`;
